From 2f0285c7ef3a3dab71e8d6f4df92ef978470c553 Mon Sep 17 00:00:00 2001
From: Xavier Claessens <xavier.claessens@collabora.com>
Date: Mon, 24 Sep 2018 09:24:38 -0400
Subject: [PATCH] Return code 0 when builddir is already configured

This is a regression in Meson 0.48.0, commit 674ae46, Meson used to
exit(0) when running setup command in a builddir already configured.
Changing to exit(1) breaks some build tools that does "meson builddir
&& ninja -C builddir".

Closes #4247.
---
 mesonbuild/mesonmain.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mesonbuild/mesonmain.py b/mesonbuild/mesonmain.py
index cd925e51f2..dfad2e7848 100644
--- a/mesonbuild/mesonmain.py
+++ b/mesonbuild/mesonmain.py
@@ -113,7 +113,7 @@ def validate_dirs(self, dir1, dir2, reconfigure):
                       '\nIf build failures persist, manually wipe your build directory to clear any\n'
                       'stored system data.\n'
                       '\nTo change option values, run "meson configure" instead.')
-                sys.exit(1)
+                sys.exit(0)
         else:
             if reconfigure:
                 print('Directory does not contain a valid build tree:\n{}'.format(build_dir))

